lib/Sema/TypeCheckType.cpp

```diff
@@ -2458,6 +2465,11 @@ Type TypeResolver::resolveDictionaryType(DictionaryTypeRepr *repr,
 
 Type TypeResolver::resolveOptionalType(OptionalTypeRepr *repr,
                                        TypeResolutionOptions options) {
+  options |= TR_OptionalTypeArgument;
```

Comment:
This should probably also apply to IUO, and it's worth checking that you *don't* get the message for something like `[@escaping () -> Void]?`.